lnwallet/channel.go

```diff
@@ -257,40 +258,54 @@ func (c *commitment) toChannelDelta(ourCommit bool) (*channeldb.ChannelDelta, er
 				if contains(dups[p.RHash], uint16(i)) {
 					continue
 				}
-
+				found = true
 				idx = uint16(i)
 				dups[p.RHash] = append(dups[p.RHash], idx)
 				break
 			}
 		}
-		return idx
+		if !found {
```

Comment:
Ok, let me know what you decide. It would have made debugging easier if it was in from the beginning. In case further changes around this code is made in the future it could potentially help, but yes, ideally we should always find the target HTLC.